mesonbuild/compilers/c.py

```diff
@@ -416,3 +421,99 @@ def get_include_args(self, path, is_system):
         if path == '':
             path = '.'
         return ['-include=' + path]
+
+
+class Xc16CCompiler(Xc16Compiler, CCompiler):
+    def __init__(self, exelist, version, for_machine: MachineChoice,
+                 is_cross, info: 'MachineInfo', exe_wrapper=None, **kwargs):
+        CCompiler.__init__(self, exelist, version, for_machine, is_cross,
+                           info, exe_wrapper, **kwargs)
+        Xc16Compiler.__init__(self)
+
+    # Override CCompiler.get_always_args
+    def get_always_args(self):
+        return ['']
+
+    def get_options(self):
+        opts = CCompiler.get_options(self)
+        opts.update({'c_std': coredata.UserComboOption('C language standard to use',
+                                                       ['none', 'c89', 'c99'],
+                                                       'none')})
+        return opts
+
+    def get_no_stdinc_args(self):
+        return []
+
+    def get_option_compile_args(self, options):
+        args = []
+        std = options['c_std']
+        if std.value == 'c89':
+            args.append('-std=gnu89')
+        elif std.value == 'c99':
+            args.append('-std=gnu99')
```

Comment:
we have gnu standards separate from c standards, does the xc16 compiler really not support c89 and c99 as defined by ISO?

Reply:
That was my mistake, I thought it was only gnu99